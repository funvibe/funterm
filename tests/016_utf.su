# UTF-8
lua.utf8_ascii = <<"Hello"/utf8>>
lua.utf8_unicode = <<"Привет"/utf8>>
lua.utf8_emoji = <<128640:32/utf8>>

# Pattern matching UTF-8
match lua.utf8_ascii {
    <<"Hello"/utf8>> -> lua.print("UTF-8 ASCII matched")
}

match lua.utf8_unicode {
    <<"Привет"/utf8>> -> lua.print("Russian UTF-8 matched")
}

match lua.utf8_emoji {
    <<char:32/utf8>> -> {
        lua.print("Emoji codepoint:", char)
    }
}

# Построение UTF строк из codepoints
lua.hello_utf8 = <<72:8/utf8, 101:8/utf8, 108:8/utf8, 108:8/utf8, 111:8/utf8>>
match lua.hello_utf8 {
    <<"Hello"/utf8>> -> lua.print("Constructed UTF-8 matches")
}
