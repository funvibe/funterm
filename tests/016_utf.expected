UTF-8 ASCII matched
Russian UTF-8 matched
Emoji codepoint: 128640
Constructed UTF-8 matches
